Loop/Managers/LoopDataManager.swift

```diff
@@ -10,150 +10,158 @@ import Foundation
 import Combine
 import HealthKit
 import LoopKit
+import LoopKitUI
 import LoopCore
 import WidgetKit
 
-protocol PresetActivationObserver: AnyObject {
-    func presetActivated(context: TemporaryScheduleOverride.Context, duration: TemporaryScheduleOverride.Duration)
-    func presetDeactivated(context: TemporaryScheduleOverride.Context)
+struct AlgorithmDisplayState {
+    var input: LoopAlgorithmInput?
+    var output: LoopAlgorithmOutput?
+
+    var activeInsulin: InsulinValue? {
+        guard let input, let value = output?.activeInsulin else {
+            return nil
+        }
+        return InsulinValue(startDate: input.predictionStart, value: value)
+    }
+
+    var activeCarbs: CarbValue? {
+        guard let input, let value = output?.activeCarbs else {
+            return nil
+        }
+        return CarbValue(startDate: input.predictionStart, value: value)
+    }
+
+    var asTuple: (algoInput: LoopAlgorithmInput?, algoOutput: LoopAlgorithmOutput?) {
+        return (algoInput: input, algoOutput: output)
+    }
+}
+
+protocol DeliveryDelegate: AnyObject {
+    var isSuspended: Bool { get }
+    var pumpInsulinType: InsulinType? { get }
+    var basalDeliveryState: PumpManagerStatus.BasalDeliveryState? { get }
+    var isPumpConfigured: Bool { get }
+
+    func enact(_ recommendation: AutomaticDoseRecommendation) async throws
+    func enactBolus(units: Double, activationType: BolusActivationType) async throws
+    func roundBasalRate(unitsPerHour: Double) -> Double
+    func roundBolusVolume(units: Double) -> Double
+}
+
+protocol DosingManagerDelegate {
+    func didMakeDosingDecision(_ decision: StoredDosingDecision)
+}
+
+enum LoopUpdateContext: Int {
+    case insulin
+    case carbs
+    case glucose
+    case preferences
+    case forecast
 }
 
+@MainActor
```

Comment:
LoopDataManager no longer has a separate data access thread, because now all operations, like loop(), and recommendation generation are done in a functional manner without state, so concurrency concerns are minimized.